src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+                    parameters.Q.Y.Length;
+
+                if (includePrivate)
+                {
+                    blobSize += parameters.D.Length;
+                }
+
+                byte[] blob = new byte[blobSize];
+                fixed (byte* pBlob = blob)
+                {
+                    // Build the header
+                    BCRYPT_ECCFULLKEY_BLOB* pBcryptBlob = (BCRYPT_ECCFULLKEY_BLOB*)pBlob;
+                    pBcryptBlob->Version = 1;
+                    pBcryptBlob->Magic = includePrivate ? KeyBlobMagicNumber.BCRYPT_ECDSA_PRIVATE_GENERIC_MAGIC : KeyBlobMagicNumber.BCRYPT_ECDSA_PUBLIC_GENERIC_MAGIC;
+                    pBcryptBlob->cbCofactor = curve.H.Length;
+                    pBcryptBlob->cbFieldLength = parameters.Q.X.Length;
+                    pBcryptBlob->cbSeed = curve.S.Length;
+                    pBcryptBlob->cbSubgroupOrder = curve.N.Length;
+                    pBcryptBlob->CurveGenerationAlgId = GetHashAlgorithm(curve.Hash);
+                    pBcryptBlob->CurveType = (Interop.BCrypt.ECC_CURVE_TYPE_ENUM)curve.CurveType;
+
+                    // Emit the blob
+                    int offset = sizeof(BCRYPT_ECCFULLKEY_BLOB);
+                    Emit(blob, ref offset, curve.P);
+                    Emit(blob, ref offset, curve.A);
+                    Emit(blob, ref offset, curve.B);
+                    Emit(blob, ref offset, curve.G.X);
+                    Emit(blob, ref offset, curve.G.Y);
+                    Emit(blob, ref offset, curve.N);
+                    Emit(blob, ref offset, curve.H);
+                    Emit(blob, ref offset, curve.S);
+                    Emit(blob, ref offset, parameters.Q.X);
+                    Emit(blob, ref offset, parameters.Q.Y);
+                    if (includePrivate)
+                    {
+                        Emit(blob, ref offset, parameters.D);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+                }
+
+                return ImportFullKeyBlob(blob, includePrivate);
+            }
+        }
+
+        #endregion
+
+        #region Export Parameters
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportExplicitCurveParameters(bool includePrivateParameters)
+        {
+            return ExportPrimeCurveParameters(this.Key, includePrivateParameters);
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportNamedCurveParameters(bool includePrivateParameters)
+        {
+            ECParameters ecparams = ExportNamedCurveParameters(this.Key, includePrivateParameters);
+            ecparams.Curve = new ECNamedCurve(new Oid(null, this.Key.GetCurveName()));
```

Comment:
I'm still confused by all of the explicitly setting of value to null.  Certainly many people might want to read oid.Value.  Since this is being returned, it should just use Oid.FromFriendlyName.